scripts/migrate-users.py

```diff
@@ -0,0 +1,237 @@
+        'passwd': django_config['database']['PASSWORD'],
+        'cursorclass': DictCursor,
+    }
+    if django_config['database'].get('HOST', None):
+        db_params['host'] = django_config['database']['HOST']
+    if django_config['database'].get('PORT', None):
+        db_params['port'] = django_config['database']['PORT']
+    django_db = MySQLdb.connect(**db_params)
+    django_cursor = django_db.cursor()
+elif django_config['database']['ENGINE'] == 'django.db.backends.sqlite3':
+    import sqlite3
+
+    # Use basic dict factory, not sqlite3.Row, for mutability
+    def dict_factory(cursor, row):
+        d = {}
+        for idx, col in enumerate(cursor.description):
+            d[col[0]] = row[idx]
+        return d
+
+    # Default location: ../codalab-worksheets/codalab/$NAME
+    sqlite_path = os.path.join(os.path.dirname(os.getcwd()),
+                               'codalab-worksheets',
+                               'codalab',
+                               django_config['database']['NAME'])
+    django_db = sqlite3.connect(sqlite_path)
+    django_db.row_factory = dict_factory
+    django_cursor = django_db.cursor()
+else:
+    raise UsageError("Invalid database engine %r" %
+                     django_config['database']['ENGINE'])
+
+# General SQL query should work on both MySQL and sqlite3
+django_cursor.execute("""
+    SELECT
+    cluser.id,
+    cluser.password,
+    cluser.last_login,
+    cluser.username AS user_name,
+    cluser.first_name,
+    cluser.last_name,
+    cluser.email,
+    cluser.date_joined,
+    cluser.organization_or_affiliation AS affiliation,
+    cluser.is_superuser,
+    email.verified AS is_verified
+    FROM authenz_cluser AS cluser
+    LEFT OUTER JOIN account_emailaddress AS email
+    ON cluser.id = email.user_id
+    WHERE cluser.id != -1 AND cluser.is_active = 1""")
+django_users = list(django_cursor.fetchall())
+
+###############################################################
+# Preprocess user data
+###############################################################
+
+# Generate new user ids using uuid
+for user in django_users:
+    user['old_user_id'] = str(user.pop('id'))
+    user['new_user_id'] = uuid.uuid4().hex
+
+###############################################################
+# Get existing users in bundles db
+###############################################################
+
+# Get set of user ids in bundles db
+with model.engine.begin() as connection:
+    bundle_users = connection.execute(select([cl_user.c.user_id])).fetchall()
+
+# Find intersection of user ids
+django_user_ids = set(user['old_user_id'] for user in django_users)
+bundle_user_ids = set(user['user_id'] for user in bundle_users)
+to_update = django_user_ids & bundle_user_ids
+to_insert = django_user_ids - bundle_user_ids
+
+print "Users to update:", ', '.join(list(to_update))
+print "Users to insert:", ', '.join(list(to_insert))
+
+to_update = [user for user in django_users
```

Comment:
Deleting users from the bundles database will not work, since there are foreign key constraints (the same reason we can't change the user IDs as you mentioned). Unfortunately it has to be separated into the insert and update steps, but makes sense to put it into one transaction.